src/main.c

```diff
@@ -165,8 +178,69 @@ static void prvTrafficGeneratorTask ( void *pvParameters )
 	}
 
 }
-static void prvTrafficLightStateTask ( void *pvParameters ){}
 
+/*-----------------------------------------------------------*/
+
+/*
+- Working on more accurately relating the green, red and traffic flow values
+- Yellow light has been defined at 2 seconds, can be changed easily but needs to be tested as well
+- Green and red values have been proportionally related
+- FreeRTOS tasks have been cleanly implemented
+- Need to make sure things work as I assume they do wrt delay functions
+
+TODO:
+- Dial in values of the red and green lights
+- Make sure the delay works properly
+- Test traffic_lights function with ADC and potentiometer
+*/
+
+static void prvTrafficLightStateTask ( void *pvParameters )
+{
+
+	SystemState systemStateToUpdate; //systemState to update
+	FlowState currentFlow; // Current traffic flow
+
+	uint16_t green_duration = 4000; // Duration of green light, also used for red light
+	uint16_t red_duration;
+
+	while(1)
+	{
+		xQueueRecieve(xFlowAdjustmentQueue, &currentFlow, 100); // Receive current Flow State
+		xQueueRecieve(xFlowAdjustmentQueue, &systemStateToUpdate, 100); // Receive current System State to update
+
+		if(systemStateToUpdate.lightState == GREEN)

```

Comment:
yeah I think this is a good start, I think there is two ways of doing this, either using software timers or vTask delay, I think we just need to update the vTaskDelay value if the traffic state changes. This seems like the most difficult part of this project, so I can also look into this with you as well.